### PR TITLE
Add a command line flag for disabling the HTTP gateway

### PR DESCRIPTION
**Pull request by raskchanky**
This should provide an escape hatch for now, until we have more granular auth.

Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/hab/src/cli.rs [hunk 1]
```diff
@@ -810,6 +810,8 @@ pub fn sub_sup_run() -> App<'static, 'static> {
             "The listen address for the gossip system [default: 0.0.0.0:9638]")
         (@arg LISTEN_HTTP: --("listen-http") +takes_value {valid_socket_addr}
             "The listen address for the HTTP Gateway [default: 0.0.0.0:9631]")
+        (@arg HTTP_DISABLE: --("http-disable") -D
+            "Disable the HTTP Gateway completely [default: enabled]")
```
Comment (baumanj) [inline components/hab/src/cli.rs]: It might be a bit clearer to write `default: false`, since `enabled` could either refer to the HTTP gateway or the `--http-disable` flag.

Alternatively, you could make this an *option* that takes a boolean rather than a flag:
```
        --http-gateway <BOOL> [default: true]
```

@apriofrost: what do you think?

File: components/sup/src/manager/mod.rs [hunk 2]
```diff
@@ -662,9 +665,13 @@ impl Manager {
         outputln!("Starting ctl-gateway on {}", &ctl_listen_addr);
         ctl_gateway::server::run(ctl_listen_addr, ctl_secret_key, ctl_tx);
         debug!("ctl-gateway started");
-        outputln!("Starting http-gateway on {}", &http_listen_addr);
-        http_gateway::Server::new(self.fs_cfg.clone(), http_listen_addr).start()?;
-        debug!("http-gateway started");
+
+        if !self.http_disable {
```
Comment (baumanj) [inline components/sup/src/manager/mod.rs]: Might be nice to do something like
```rust
        if self.http_disable {
            info!("http-gateway disabled");
        else {
            …
```
This adds some useful information and avoids the `!disable`; I find double-negations a little bit hard to parse.

File: components/hab/src/cli.rs (hunk 1 above)
Comment (raskchanky) [reply components/hab/src/cli.rs]: As do I - I'll change it.

File: components/sup/src/manager/mod.rs (hunk 2 above)
Comment (raskchanky) [reply components/sup/src/manager/mod.rs]: Yeah, fair point.  I'll add that.

**Comment by christophermaier** [comment]
Going to go ahead and merge. @apriofrost is coming up-to-speed on the CLI and will give more holistic feedback once she's there.
